lib/private/Server.php

```diff
@@ -974,6 +974,15 @@ public function __construct($webRoot, \OC\Config $config) {
 				$stream = 'php://input';
 			}
 
+			if(isset($_SERVER) && isset($_SERVER['REQUEST_METHOD']) && $_SERVER['REQUEST_METHOD'] === "POST") {
+				foreach($_SERVER as $item => $value) {
+					if($item === "HTTP_X_WEBDAV") {
+						$_SERVER["REQUEST_METHOD"]=$value;
+						break;
+					}
+				};
```

Comment:
Why are you iterating over $_SERVER when we are able to directly access the key?

```suggestion
				if (!empty($_SERVER['HTTP_X_WEBDAV'])) {
					$_SERVER['REQUEST_METHOD'] = $_SERVER['HTTP_X_WEBDAV'];
				}
```